core/src/main/java/org/bf2/srs/fleetmanager/service/impl/FileQuotaPlansService.java

```diff
@@ -111,10 +124,75 @@ public void init() throws IOException {
             tmClient.validateConfig(p.getResources());
             plans.put(p.getName(), p);
         }
+
+        List<OrganizationAssignment> staticOrganizationAssignments = quotaPlansConfigList.getOrganizations();
+        if (staticOrganizationAssignments == null)
+            staticOrganizationAssignments = Collections.emptyList();
+
+        for (OrganizationAssignment assignment : staticOrganizationAssignments) {
+            if (!plans.containsKey(assignment.getPlan())) {
+                throw new IllegalStateException("Could not find quota plan named '" + assignment.getPlan() +
+                        "' intended for organization ID '" + assignment.getOrgId() + "'");
+            }
+            organizationAssignments.put(assignment.getOrgId(), assignment);
+        }
+
+        if (quotaPlansConfigList.getReconcile() != null && quotaPlansConfigList.getReconcile()) {
+            reconcile();
+        }
+    }
+
+    private void reconcile() {
+        log.info("Performing quota plan reconciliation");
+        var allRegistries = storage.getAllRegistries();
+        var updatedCount = 0;
+        for (RegistryData registry : allRegistries) {
+            var tid = registry.getId();
+            var tmc = Utils.createTenantManagerConfig(registry.getRegistryDeployment());
+            try {
+                var tenant = tmClient.getTenantById(tmc, tid).orElseThrow();
+                Map<String, Long> tenantLimits = new HashMap<>();
+                for (TenantLimit resource : tenant.getResources()) {
+                    tenantLimits.put(resource.getType(), resource.getLimit());
+                }
+
+                var targetPlan = determineQuotaPlan(registry.getOrgId());
+
+                var requiresUpdate = false;
+                // Compare limits
+                for (TenantLimit targetLimit : targetPlan.getResources()) {
+                    var v = tenantLimits.get(targetLimit.getType());
+                    if (v == null || !v.equals(targetLimit.getLimit())) {
+                        requiresUpdate = true;
+                        break;
+                    }
+                }
+                if (requiresUpdate) {
+                    UpdateTenantRequest utr = UpdateTenantRequest.builder()
+                            .id(tid)
+                            .status(tenant.getStatus())
+                            .resources(targetPlan.getResources())
+                            .build();
+                    tmClient.updateTenant(tmc, utr);
+                    updatedCount++;
+                }
+
+            } catch (TenantManagerServiceException | NoSuchElementException | TenantNotFoundServiceException e) {
```

Comment:
Do we really need to catch NoSuchElementException? Are we missing something before that prevent that exception from being thrown?

Reply:
It's thrown from here https://github.com/bf2fc6cc711aee1a0c2a/srs-fleet-manager/pull/252/files#diff-7c044f16023437ab8de098b9e5729c439704a8f29fc8d44ebfa8bc76ea342b41R153 , I can handle that explicitly, but the result would be the same.

Reply:
That was my point. You can pass a supplier providing a more specific exception to throw instead of throwing this one.